### PR TITLE
[FIX] Evaluation Results input validation

### PR DESCRIPTION
**Pull request by ales-erjavec**
##### Issue

Most widget in evaluation category fail when presented either an empty Evaluation Results instance or one containing nan values. The source of these is in most cases the 'Predictions' widget.

##### Description of changes

* 'Predictions' widget filters out unknown values from the 'Evaluation Results' when the corresponding true target value is unknown
* Confusion Matrix, ROC Analysis, Lift Curve and Calibration plot widgets handle empty results
* Confusion Matrix, ROC Analysis, Lift Curve and Calibration plot widgets check for the presence of Nan values and bail out early with an error message.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ales-erjavec** [comment]
https://sentry.io/biolab/orange3/issues/204404003/

**Comment by ales-erjavec** [comment]
https://sentry.io/biolab/orange3/issues/208378492/

**Comment by ales-erjavec** [comment]
https://sentry.io/biolab/orange3/issues/208398496/

**Comment by ales-erjavec** [comment]
https://sentry.io/biolab/orange3/issues/208526017/

**Comment by ales-erjavec** [comment]
https://sentry.io/biolab/orange3/issues/207080848/

**Comment by ales-erjavec** [comment]
https://sentry.io/biolab/orange3/issues/196743961/

File: Orange/widgets/evaluate/owconfusionmatrix.py [hunk 1]
```diff
@@ -243,6 +245,21 @@ def set_results(self, results):
...
+                    numpy.any(numpy.isnan(results.predicted)):
+                # Error out here (could filter them out with a warning
+                # instead).
+                nan_values = True
+                results = data = None
+
+        if nan_values:
+            self.Error.invalid_values()
+        else:
+            self.Error.invalid_values.clear()
```
Comment (janezd) [inline Orange/widgets/evaluate/owconfusionmatrix.py]: No need to change anything here, but just if you didn't know: if you prefer, you can use `self.Error.invalid_values(shown=nan_values)`, which shows or hides the messages.

**Pull request merged**